src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,6 +73,36 @@ public static void ResetColor()
             }
         }
 
+        public static int WindowWidth
+        {
+            get
+            {
+                if (!ConsoleOutIsTerminal) return 0; // Should we rather throw exception here? On windows we get IOException from the native code, do we need to match for code portability?
+                return TerminalBasicInfo.Instance.ColumnFormat;
```

Comment:
We strive to use things that are POSIX compliant, but in this case there isn't, so we do the best we can.  I'd suggest doing adding something to the native shim that uses ioctl with TIOCGWINSZ  if it's available, or else returns an error indicating it's not supported, and we can either throw a PlatformNotSupported or return a sentiinel value from WindowSize.  That should work for both Linux and OS X, and also FreeBSD.  The native shim's implementation can be augmented in the future with other implementations as we find it necessary.